ddp-workspace/projects/ddp-dsm-ui/src/app/phi-manifest/phi-manifest.component.ts

```diff
@@ -0,0 +1,112 @@
+import {Component, OnInit} from '@angular/core';
+import {Auth} from '../services/auth.service';
+import {ComponentService} from '../services/component.service';
+import {DSMService} from '../services/dsm.service';
+import {Statics} from '../utils/statics';
+import {Utils} from '../utils/utils';
+import {ActivatedRoute} from '@angular/router';
+import {PhiManifestResponseModel} from './phi-manifest-response.model';
+import {PhiManifestModel} from './phi-manifest.model';
+
+@Component({
+    selector: 'app-phi-manifest',
+    templateUrl: './phi-manifest.component.html',
+    styleUrls: ['./phi-manifest.component.scss']
+})
+
+export class PhiManifestComponent implements OnInit {
+    errorMessage: string;
+    additionalMessage: string;
+    loading = false;
+    realm: string;
+    participantId: string = null;
+    sequencingOrderId: string = null;
+    allowedToSeeInformation = false;
+
+    constructor(private dsmService: DSMService, private auth: Auth, private route: ActivatedRoute) {
+        if (!auth.authenticated()) {
+            auth.sessionLogout();
+        }
+        this.route.queryParams.subscribe(params => {
+            this.realm = params[DSMService.REALM] || null;
+            if (this.realm != null) {
+                this.checkRight();
+            }
+        });
+    }
+
+    private checkRight(): void {
+        this.allowedToSeeInformation = false;
+        this.additionalMessage = null;
+        let jsonData: any[];
+        this.dsmService.getRealmsAllowed(Statics.MEDICALRECORD).subscribe({
+            next: data => {
+                jsonData = data;
+                jsonData.forEach((val) => {
+                    if (this.realm === val) {
+                        this.allowedToSeeInformation = true;
+                    }
+                });
+                if (!this.allowedToSeeInformation) {
+                    this.additionalMessage = 'You are not allowed to see information of the selected study at that category';
+                }
+            },
+            error: () => null
+        });
+    }
+
+    ngOnInit(): void {
+        if (sessionStorage.getItem(ComponentService.MENU_SELECTED_REALM) != null) {
+            this.realm = sessionStorage.getItem(ComponentService.MENU_SELECTED_REALM);
+            this.checkRight();
+        } else {
+            this.additionalMessage = 'Please select a PE-CGS study';
+        }
+        window.scrollTo(0, 0);
+    }
+
+    getPhiReport(): void {
```

Comment:
`get()` methods should return something, but this is void.  Maybe it should be `downloadManifest()`?